tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
-		peerOrg := config.PeerOrganizations[i]
-		numComponents = peerOrg.NumPeers
-		err = k.deleteConfigMaps(numComponents, "peer", peerOrg.Name, config.TLS, "configmaps")
-		if err != nil {
-			logger.ERROR("Failed to delete peer secrets in %s", peerOrg.Name)
-		}
+	err = yaml.Unmarshal(contents, &coreConfig)
+	if err != nil {
+		return coreConfig, err
 	}
-	k8sServicesFile := paths.ConfigFilePath("services")
-	k8sPodsFile := paths.ConfigFilePath("pods")
-
-	var inputPaths []string
-	if config.K8s.DataPersistence == "local" {
-		inputPaths = []string{k.dataPersistenceFilePath(config)}
-		k = K8s{KubeConfigPath: kubeConfigPath, Action: "apply", Arguments: inputPaths}
-		_, err = networkclient.ExecuteK8sCommand(k.Args(), true)
-		if err != nil {
-			logger.ERROR("Failed to launch k8s pod")
-		}
+	coreConfig.VM.Endpoint = "localhost:2375"
+	coreConfig.Peer.ChaincodeListenAddress = "0.0.0.0:7052"
+	if nsConfig.GossipEnable {
+		coreConfig.Peer.Gossip.State.Enabled = true
+		coreConfig.Peer.Gossip.UseLeaderElection = true
+		coreConfig.Peer.Gossip.OrgLeader = false
+	} else {
+		coreConfig.Peer.Gossip.State.Enabled = false
+		coreConfig.Peer.Gossip.UseLeaderElection = false
+		coreConfig.Peer.Gossip.OrgLeader = true
+	}
+	if nsConfig.TLS == "true" || nsConfig.TLS == "mutual" {
+		coreConfig.Peer.TLS.Enabled = true
+	} else {
+		coreConfig.Peer.TLS.Enabled = false
+	}
+	coreConfig.Peer.TLS.Cert.File = "/etc/hyperledger/fabric/artifacts/tls/server.crt"
+	coreConfig.Peer.TLS.Key.File = "/etc/hyperledger/fabric/artifacts/tls/server.key"
+	coreConfig.Peer.ChaincodeAddress = "localhost:7052"
+	ccExecuteTimeout, _ := time.ParseDuration("1500s")
+	coreConfig.Chaincode.ExecuteTimeout = ccExecuteTimeout
+	coreConfig.Peer.MSPConfigPath = "/etc/hyperledger/fabric/artifacts/msp"
+	coreConfig.Peer.FileSystemPath = "/shared/data"
+	if nsConfig.Metrics {
+		coreConfig.Operations.TLS.Enabled = false
+		coreConfig.Metrics.Provider = "prometheus"
 	}
-	inputPaths = []string{k8sServicesFile, k8sPodsFile}
-	if config.K8s.DataPersistence == "true" || config.K8s.DataPersistence == "local" {
-		inputPaths = append(inputPaths, k.dataPersistenceFilePath(config))
+	if nsConfig.DBType == "couchdb" {
+		coreConfig.Ledger.State.StateDatabase = "CouchDB"
+		coreConfig.Ledger.State.CouchDBConfig.CouchDBAddress = "localhost:5984"
 	}
+	coreConfig.Chaincode.Builder = k8s.dockerImage("ccenv", nsConfig.DockerOrg, nsConfig.DockerTag, nsConfig.DockerImages.Ccenv)
+	coreConfig.Chaincode.Golang.Runtime = k8s.dockerImage("baseos", nsConfig.DockerOrg, nsConfig.DockerTag, nsConfig.DockerImages.Baseos)
+	coreConfig.Chaincode.Java.Runtime = k8s.dockerImage("javaenv", nsConfig.DockerOrg, nsConfig.DockerTag, nsConfig.DockerImages.Javaenv)
+	coreConfig.Chaincode.Node.Runtime = k8s.dockerImage("nodeenv", nsConfig.DockerOrg, nsConfig.DockerTag, nsConfig.DockerImages.Nodeenv)
+	return coreConfig, nil
+}
+
+func (k8s K8s) orderer(nsConfig networkspec.Config) (fabricconfig.Orderer, error) {
```

Comment:
Can we push this up into the `fabricconfig/orderer.go`? Or is there something that would keep us from putting it there?

Reply:
yes agree. moved to orderer.go.